graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -466,18 +455,22 @@ public boolean removeStream(Stream stream) {
         return ImmutableSet.copyOf(this.indexSets);
     }
 
+    @SuppressWarnings("unchecked")
     public List<String> getStreamIds() {
-        if (!hasField(FIELD_STREAMS)) {
-            return streams.stream().map(Stream::getId).collect(Collectors.toList());
-        }
+        List<String> streamField;
         try {
-            @SuppressWarnings("unchecked")
-            final List<String> streamIds = getFieldAs(List.class, FIELD_STREAMS);
-            return new ArrayList<>(streamIds);
+            streamField = getFieldAs(List.class, FIELD_STREAMS);
```

Comment:
I found out why this still fails with the archive restore. The archive creates a message with a `Set<String>` for the streams field in the map.